### PR TITLE
feat(cli): SSR resolution externals

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- Pull in some minor changes from the RSC branch.
- Add microsecond formatting to the terminal reporter.
- Add externals to prevent duplicate copies of source-map-explorer from breaking source maps in the CLI during API Route + RSC development.
- Add support for resolving native packages in a server context.
- Lazily search for asset registry path, and use the correct version from the metro config.
- Simplify alias of asset registry path on web.
- Improve logging for API routes.

# Test Plan

- Added new resolver unit tests.

### CODE REVIEW

File: packages/@expo/cli/src/start/server/metro/MetroBundlerDevServer.ts [hunk 1]
```diff
@@ -910,6 +932,13 @@ export class MetroBundlerDevServer extends BundlerDevServer {
     this.pendingRouteOperations.clear();
   }
 
+  // Metro HMR
+
+  private setupHmr(url: URL) {
+    debug('[CLI]: Register SSR HMR bundle URL:', url.toString());
+    // TODO: Pull in HMR SSR code.
```
Comment (byCedric) [inline packages/@expo/cli/src/start/server/metro/MetroBundlerDevServer.ts:939]: What needs to happen for HMR & SSR?

File: packages/@expo/cli/src/start/server/metro/withMetroMultiPlatform.ts [hunk 2]
```diff
@@ -350,6 +413,42 @@ export function withExtendedResolver(
       };
     },
 
+    // Custom externals support
+    (context: ResolutionContext, moduleName: string, platform: string | null) => {
+      // We don't support this in the resolver at the moment.
+      if (moduleName.endsWith('/package.json')) {
+        return null;
+      }
```
Comment (byCedric) [inline packages/@expo/cli/src/start/server/metro/withMetroMultiPlatform.ts:421]: Do we need to support this eventually?

**Review by byCedric — APPROVED** [review]

File: packages/@expo/cli/src/start/server/metro/withMetroMultiPlatform.ts (hunk 2 above)
Comment (byCedric) [reply packages/@expo/cli/src/start/server/metro/withMetroMultiPlatform.ts:421]: As in, it's not really a TODO item

File: packages/@expo/cli/src/start/server/metro/MetroBundlerDevServer.ts (hunk 1 above)
Comment (EvanBacon) [reply packages/@expo/cli/src/start/server/metro/MetroBundlerDevServer.ts:939]: A self-contained branch must be merged https://github.com/expo/expo/compare/%40evanbacon/cli/ssr-hmr-stub?expand=1

File: packages/@expo/cli/src/start/server/metro/withMetroMultiPlatform.ts (hunk 2 above)
Comment (EvanBacon) [reply packages/@expo/cli/src/start/server/metro/withMetroMultiPlatform.ts:421]: We may need to support it in the future.

**Pull request merged**